hydra-node/exe/hydra-node/Main.hs

```diff
@@ -110,6 +126,29 @@ main = do
 
     action (Ledger.cardanoLedger globals ledgerEnv)
 
+  -- check if hydra-node parameters are matching with the hydra-node state.
+  -- REVIEW: Should we also check against items we receive on-chain here? e.g. 'OpenThreadOutput' if in Open state?
```

Comment:
I wouldn't try to cover for everything in this P.R. I would focus on operations errors and especially the case where an operator would change the node configuration and restart it even though a head was already open and is not compatible with this configuration.

So I wouldn't check against items received on-chain as I don't expect them to be related to any operator _error_. Or maybe I'm missing something.